diff --git a/cl_dll/StudioModelRenderer.cpp b/cl_dll/StudioModelRenderer.cpp
index 45e9f66..27b2b99 100644
--- a/cl_dll/StudioModelRenderer.cpp
+++ b/cl_dll/StudioModelRenderer.cpp
@@ -27,6 +27,8 @@
 #include "StudioModelRenderer.h"
 #include "GameStudioModelRenderer.h"
 
+#include "byteswap.h"
+
 // Global engine <-> studio model rendering code interface
 engine_studio_api_t IEngineStudio;
 
@@ -213,30 +215,30 @@ void CStudioModelRenderer::StudioCalcBoneQuaterion( int frame, float s, mstudiob
 			// Bah, missing blend!
 			if( panimvalue->num.valid > k )
 			{
-				angle1[j] = panimvalue[k + 1].value;
+				angle1[j] = LittleShort(panimvalue[k + 1].value);
 
 				if( panimvalue->num.valid > k + 1 )
 				{
-					angle2[j] = panimvalue[k + 2].value;
+					angle2[j] = LittleShort(panimvalue[k + 2].value);
 				}
 				else
 				{
 					if( panimvalue->num.total > k + 1 )
 						angle2[j] = angle1[j];
 					else
-						angle2[j] = panimvalue[panimvalue->num.valid + 2].value;
+						angle2[j] = LittleShort(panimvalue[panimvalue->num.valid + 2].value);
 				}
 			}
 			else
 			{
-				angle1[j] = panimvalue[panimvalue->num.valid].value;
+				angle1[j] = LittleShort(panimvalue[panimvalue->num.valid].value);
 				if( panimvalue->num.total > k + 1 )
 				{
 					angle2[j] = angle1[j];
 				}
 				else
 				{
-					angle2[j] = panimvalue[panimvalue->num.valid + 2].value;
+					angle2[j] = LittleShort(panimvalue[panimvalue->num.valid + 2].value);
 				}
 			}
 			angle1[j] = pbone->value[j+3] + angle1[j] * pbone->scale[j + 3];
@@ -303,11 +305,11 @@ void CStudioModelRenderer::StudioCalcBonePosition( int frame, float s, mstudiobo
 				// and there's more data in the span
 				if( panimvalue->num.valid > k + 1 )
 				{
-					pos[j] += ( panimvalue[k + 1].value * ( 1.0f - s ) + s * panimvalue[k + 2].value ) * pbone->scale[j];
+					pos[j] += ( LittleShort(panimvalue[k + 1].value) * ( 1.0f - s ) + s * LittleShort(panimvalue[k + 2].value) ) * pbone->scale[j];
 				}
 				else
 				{
-					pos[j] += panimvalue[k + 1].value * pbone->scale[j];
+					pos[j] += LittleShort(panimvalue[k + 1].value) * pbone->scale[j];
 				}
 			}
 			else
@@ -315,11 +317,11 @@ void CStudioModelRenderer::StudioCalcBonePosition( int frame, float s, mstudiobo
 				// are we at the end of the repeating values section and there's another section with data?
 				if( panimvalue->num.total <= k + 1 )
 				{
-					pos[j] += ( panimvalue[panimvalue->num.valid].value * ( 1.0f - s ) + s * panimvalue[panimvalue->num.valid + 2].value ) * pbone->scale[j];
+					pos[j] += ( LittleShort(panimvalue[panimvalue->num.valid].value) * ( 1.0f - s ) + s * LittleShort(panimvalue[panimvalue->num.valid + 2].value) ) * pbone->scale[j];
 				}
 				else
 				{
-					pos[j] += panimvalue[panimvalue->num.valid].value * pbone->scale[j];
+					pos[j] += LittleShort(panimvalue[panimvalue->num.valid].value) * pbone->scale[j];
 				}
 			}
 		}
@@ -404,8 +406,26 @@ mstudioanim_t *CStudioModelRenderer::StudioGetAnim( model_t *m_pSubModel, mstudi
 
 	if( !IEngineStudio.Cache_Check( (struct cache_user_s *)&( paSequences[pseqdesc->seqgroup] ) ) )
 	{
+		mstudioanim_t *panim;
+		mstudioseqdesc_t *pseqdesc2 = (mstudioseqdesc_t *)((byte *)m_pStudioHeader + m_pStudioHeader->seqindex);
 		gEngfuncs.Con_DPrintf("loading %s\n", pseqgroup->name );
 		IEngineStudio.LoadCacheFile( pseqgroup->name, (struct cache_user_s *)&paSequences[pseqdesc->seqgroup] );
+
+		// all anim offsets in this seqgroup *must* be swapped, which is a bit evil
+		
+		for (int i = 0; i < m_pStudioHeader->numseq; i++) {
+			if (pseqdesc2[i].seqgroup == pseqdesc->seqgroup) {
+				panim = (mstudioanim_t *)((byte *)paSequences[pseqdesc->seqgroup].data + pseqdesc2[i].animindex);
+				for (int j = 0; j < pseqdesc2[i].numblends; j++) {
+					for (int k = 0; k < m_pStudioHeader->numbones; k++) {
+						for (int l = 0; l < 6; l++) {
+							LittleShortSW(panim->offset[l]);
+						}
+						panim++;
+					}
+				}
+			}
+		}
 	}
 	return (mstudioanim_t *)( (byte *)paSequences[pseqdesc->seqgroup].data + pseqdesc->animindex );
 }
diff --git a/dlls/nodes.cpp b/dlls/nodes.cpp
index 21d759d..393261d 100644
--- a/dlls/nodes.cpp
+++ b/dlls/nodes.cpp
@@ -24,6 +24,7 @@
 #include	"nodes_compat.h"
 #include	"animation.h"
 #include	"doors.h"
+#include	"byteswap.h"
 
 #define	HULL_STEP_SIZE 16// how far the test hull moves on each step
 #define	NODE_HEIGHT	8	// how high to lift nodes off the ground after we drop them all (make stair/ramp mapping easier)
@@ -2396,7 +2397,7 @@ int CGraph::FLoadGraph( const char *szMapName )
 	length -= sizeof(int);
 	if( length < 0 )
 		goto ShortFile;
-	iVersion = *(int *) pMemFile;
+	iVersion = LittleLong(*(int *) pMemFile);
 	pMemFile += sizeof(int);
 
 	if( iVersion == GRAPH_VERSION || iVersion == GRAPH_VERSION_RETAIL )
@@ -2418,6 +2419,8 @@ int CGraph::FLoadGraph( const char *szMapName )
 			m_di = NULL;
 			m_pRouteInfo = NULL;
 			m_pHashLinks = NULL;
+
+			ByteswapGraph();
 		}
 #if _GRAPH_VERSION != _GRAPH_VERSION_RETAIL
 		else
@@ -2449,6 +2452,8 @@ int CGraph::FLoadGraph( const char *szMapName )
 		memcpy( m_pNodes, pMemFile, sizeof(CNode) * m_cNodes );
 		pMemFile += sizeof(CNode) * m_cNodes;
 
+		ByteswapNodes();
+
 		// Malloc for the link pool
 		//
 		m_pLinkPool = (CLink *)calloc( sizeof(CLink), m_cLinks );
@@ -2468,6 +2473,8 @@ int CGraph::FLoadGraph( const char *szMapName )
 				goto ShortFile;
 			memcpy( m_pLinkPool, pMemFile, sizeof(CLink) * m_cLinks );
 			pMemFile += sizeof(CLink) * m_cLinks;
+
+			ByteswapLinks();
 		}
 #if _GRAPH_VERSION != _GRAPH_VERSION_RETAIL
 		else
@@ -2501,6 +2508,8 @@ int CGraph::FLoadGraph( const char *szMapName )
 		memcpy( m_di, pMemFile, sizeof(DIST_INFO) * m_cNodes );
 		pMemFile += sizeof(DIST_INFO) * m_cNodes;
 
+		ByteswapDistInfo();
+
 		// Malloc for the routing info.
 		//
 		m_fRoutingComplete = FALSE;
@@ -2542,6 +2551,8 @@ int CGraph::FLoadGraph( const char *szMapName )
 		memcpy( m_pHashLinks, pMemFile, sizeof(short) * m_nHashLinks );
 		// pMemFile += sizeof(short) * m_nHashLinks;
 
+		ByteswapHashLinks();
+
 		// Set the graph present flag, clear the pointers set flag
 		//
 		m_fGraphPresent = TRUE;
@@ -2576,7 +2587,7 @@ NoMemory:
 //=========================================================
 int CGraph::FSaveGraph( const char *szMapName )
 {
-	int iVersion = GRAPH_VERSION;
+	int iVersion = LittleLong(GRAPH_VERSION);
 	char szFilename[MAX_PATH];
 	FILE *file;
 
@@ -2613,17 +2624,30 @@ int CGraph::FSaveGraph( const char *szMapName )
 		// write the version
 		fwrite( &iVersion, sizeof(int), 1, file );
 
+		ByteswapGraph();
+
 		// write the CGraph class
 		fwrite( this, sizeof(CGraph), 1, file );
 
+		ByteswapGraph();
+		ByteswapNodes();
+
 		// write the nodes
 		fwrite( m_pNodes, sizeof(CNode), m_cNodes, file );
 
+		ByteswapNodes();
+		ByteswapLinks();
+
 		// write the links
 		fwrite( m_pLinkPool, sizeof(CLink), m_cLinks, file );
 
+		ByteswapLinks();
+		ByteswapDistInfo();
+
 		fwrite( m_di, sizeof(DIST_INFO), m_cNodes, file );
 
+		ByteswapDistInfo();
+
 		// Write the route info.
 		//
 		if( m_pRouteInfo && m_nRouteInfo )
@@ -2633,7 +2657,11 @@ int CGraph::FSaveGraph( const char *szMapName )
 
 		if( m_pHashLinks && m_nHashLinks )
 		{
+			ByteswapHashLinks();
+
 			fwrite( m_pHashLinks, sizeof(short), m_nHashLinks, file );
+
+			ByteswapHashLinks();
 		}
 		fclose( file );
 		return TRUE;
@@ -3563,6 +3591,91 @@ EnoughSaid:
 	pMyPath2 = 0;
 }
 
+void CGraph::ByteswapGraph()
+{
+	LittleLongSW(m_cNodes);
+	LittleLongSW(m_cLinks);
+	LittleLongSW(m_nRouteInfo);
+
+	for (int i = 0; i < 3; i++)
+	{
+		for (int j = 0; j < NUM_RANGES; j++)
+		{
+			LittleLongSW(m_RangeStart[i][j]);
+			LittleLongSW(m_RangeEnd[i][j]);
+		}
+		
+		m_RegionMin[i] = LittleFloat(m_RegionMin[i]);
+		m_RegionMax[i] = LittleFloat(m_RegionMax[i]);
+	}
+
+	LittleLongSW(m_nHashLinks);
+
+	for (int i = 0; i < 16; i++)
+	{
+		LittleLongSW(m_HashPrimes[i]);
+	}
+}
+
+void CGraph::ByteswapNodes()
+{
+	for (int i = 0; i < m_cNodes; i++)
+	{
+		m_pNodes[i].m_vecOrigin.x = LittleFloat(m_pNodes[i].m_vecOrigin.x);
+		m_pNodes[i].m_vecOrigin.y = LittleFloat(m_pNodes[i].m_vecOrigin.y);
+		m_pNodes[i].m_vecOrigin.z = LittleFloat(m_pNodes[i].m_vecOrigin.z);
+		m_pNodes[i].m_vecOriginPeek.x = LittleFloat(m_pNodes[i].m_vecOriginPeek.x);
+		m_pNodes[i].m_vecOriginPeek.y = LittleFloat(m_pNodes[i].m_vecOriginPeek.y);
+		m_pNodes[i].m_vecOriginPeek.z = LittleFloat(m_pNodes[i].m_vecOriginPeek.z);
+		LittleLongSW(m_pNodes[i].m_afNodeInfo);
+		LittleLongSW(m_pNodes[i].m_cNumLinks);
+		LittleLongSW(m_pNodes[i].m_iFirstLink);
+		
+		for (int j = 0; j < MAX_NODE_HULLS; j++)
+		{
+			LittleLongSW(m_pNodes[i].m_pNextBestNode[j][0]);
+			LittleLongSW(m_pNodes[i].m_pNextBestNode[j][1]);
+		}
+		
+		m_pNodes[i].m_flClosestSoFar = LittleFloat(m_pNodes[i].m_flClosestSoFar);
+		LittleLongSW(m_pNodes[i].m_iPreviousNode);
+		LittleShortSW(m_pNodes[i].m_sHintType);
+		LittleShortSW(m_pNodes[i].m_sHintActivity);
+		m_pNodes[i].m_flHintYaw = LittleFloat(m_pNodes[i].m_flHintYaw);
+	}
+}
+
+void CGraph::ByteswapLinks()
+{
+	for (int i = 0; i < m_cLinks; i++)
+	{
+		LittleLongSW(m_pLinkPool[i].m_iSrcNode);
+		LittleLongSW(m_pLinkPool[i].m_iDestNode);
+		LittleLongSW(m_pLinkPool[i].m_afLinkInfo);
+		m_pLinkPool[i].m_flWeight = LittleFloat(m_pLinkPool[i].m_flWeight);
+	}
+}
+
+void CGraph::ByteswapDistInfo()
+{
+	for (int i = 0; i < m_cNodes; i++)
+	{
+		LittleLongSW(m_di[i].m_SortedBy[0]);
+		LittleLongSW(m_di[i].m_SortedBy[1]);
+		LittleLongSW(m_di[i].m_SortedBy[2]);
+		LittleLongSW(m_di[i].m_CheckedEvent);
+	}
+}
+
+void CGraph::ByteswapHashLinks()
+{
+	for (int i = 0; i < m_nHashLinks; i++)
+	{
+		LittleShortSW(m_pHashLinks[i]);
+	}
+}
+
+
 //=========================================================
 // CNodeViewer - Draws a graph of the shorted path from all nodes
 // to current location (typically the player).  It then draws
diff --git a/dlls/nodes.h b/dlls/nodes.h
index becfd07..4c98b29 100644
--- a/dlls/nodes.h
+++ b/dlls/nodes.h
@@ -212,6 +212,11 @@ public:
 		return 0; 
 	}
 
+	void	ByteswapGraph();
+	void	ByteswapNodes();
+	void	ByteswapLinks();
+	void	ByteswapDistInfo();
+	void	ByteswapHashLinks();
 
 	inline	CNode &Node( int i )
 	{
diff --git a/dlls/nodes_compat.h b/dlls/nodes_compat.h
index c190f11..2c994d5 100644
--- a/dlls/nodes_compat.h
+++ b/dlls/nodes_compat.h
@@ -8,6 +8,7 @@
 #if _GRAPH_VERSION != _GRAPH_VERSION_RETAIL
 
 #include "stdint.h"
+#include "byteswap.h"
 
 typedef int32_t PTR32;
 
@@ -57,15 +58,32 @@ public:
 		other->m_pLinkPool = NULL;
 		other->m_pRouteInfo = NULL;
 
-		other->m_cNodes	= m_cNodes;
-		other->m_cLinks	= m_cLinks;
-		other->m_nRouteInfo	= m_nRouteInfo;
+		other->m_cNodes	= LittleLong(m_cNodes);
+		other->m_cLinks	= LittleLong(m_cLinks);
+		other->m_nRouteInfo	= LittleLong(m_nRouteInfo);
 
 		other->m_di = NULL;
 
 		memcpy( (void *) &other->m_RangeStart, (void *) m_RangeStart,
 				offsetof(class CGraph, m_pHashLinks) - offsetof(class CGraph, m_RangeStart) );
 
+		for (int i = 0; i < 3; i++)
+		{
+			for (int j = 0; j < NUM_RANGES; j++)
+			{
+				LittleLongSW(other->m_RangeStart[i][j]);
+				LittleLongSW(other->m_RangeEnd[i][j]);
+			}
+			
+			other->m_RegionMin[i] = LittleFloat(other->m_RegionMin[i]);
+			other->m_RegionMax[i] = LittleFloat(other->m_RegionMax[i]);
+		}
+
+		for (int i = 0; i < 16; i++)
+		{
+			LittleLongSW(other->m_HashPrimes[i]);
+		}
+
 #if 0	          // replacement routine in case a change in CGraph breaks the above memcpy
 		for (int i = 0; i < 3; ++i)
 			for (int j = 0; j < NUM_RANGES; ++j)
@@ -105,7 +123,7 @@ public:
 #endif
 
 		other->m_pHashLinks = NULL;
-		other->m_nHashLinks	= m_nHashLinks;
+		other->m_nHashLinks	= LittleLong(m_nHashLinks);
 
 		other->m_iLastActiveIdleSearch	= m_iLastActiveIdleSearch;
 
@@ -128,14 +146,14 @@ public:
 	float	m_flWeight;
 
 	void copyOverTo(CLink* other) {
-		other->m_iSrcNode	= m_iSrcNode;
-		other->m_iDestNode	= m_iDestNode	;
+		other->m_iSrcNode	= LittleLong(m_iSrcNode);
+		other->m_iDestNode	= LittleLong(m_iDestNode);
 		other->m_pLinkEnt	= NULL;
 		for (int i = 0; i < 4; ++i)
 			other->m_szLinkEntModelname[i]	= m_szLinkEntModelname[i];
 //		m_szLinkEntModelname[ 4 ]
-		other->m_afLinkInfo	= m_afLinkInfo;
-		other->m_flWeight	= m_flWeight;
+		other->m_afLinkInfo	= LittleLong(m_afLinkInfo);
+		other->m_flWeight	= LittleFloat(m_flWeight);
 	}
 };
 
diff --git a/dlls/saverestore.h b/dlls/saverestore.h
index 993dead..420e416 100644
--- a/dlls/saverestore.h
+++ b/dlls/saverestore.h
@@ -53,16 +53,16 @@ class CSave : public CSaveRestoreBuffer
 public:
 	CSave( SAVERESTOREDATA *pdata ) : CSaveRestoreBuffer( pdata ) {};
 
-	void	WriteShort( const char *pname, const short *value, int count );
-	void	WriteInt( const char *pname, const int *value, int count );		// Save an int
-	void	WriteFloat( const char *pname, const float *value, int count );	// Save a float
-	void	WriteTime( const char *pname, const float *value, int count );	// Save a float (timevalue)
+	void	WriteShort( const char *pname, short *value, int count );
+	void	WriteInt( const char *pname, int *value, int count );		// Save an int
+	void	WriteFloat( const char *pname, float *value, int count );	// Save a float
+	void	WriteTime( const char *pname, float *value, int count );	// Save a float (timevalue)
 	void	WriteData( const char *pname, int size, const char *pdata );		// Save a binary data block
 	void	WriteString( const char *pname, const char *pstring );			// Save a null-terminated string
 	void	WriteString( const char *pname, const int *stringId, int count );	// Save a null-terminated string (engine string)
-	void	WriteVector( const char *pname, const Vector &value );				// Save a vector
-	void	WriteVector( const char *pname, const float *value, int count );	// Save a vector
-	void	WritePositionVector( const char *pname, const Vector &value );		// Offset for landmark if necessary
+	void	WriteVector( const char *pname, Vector &value );				// Save a vector
+	void	WriteVector( const char *pname, float *value, int count );	// Save a vector
+	void	WritePositionVector( const char *pname, Vector &value );		// Offset for landmark if necessary
 	void	WritePositionVector( const char *pname, const float *value, int count );	// array of pos vectors
 	void	WriteFunction( const char *pname, void **value, int count );		// Save a function pointer
 	int		WriteEntVars( const char *pname, entvars_t *pev );		// Save entvars_t (entvars_t)
diff --git a/dlls/util.cpp b/dlls/util.cpp
index 1427b9e..05af6ef 100644
--- a/dlls/util.cpp
+++ b/dlls/util.cpp
@@ -30,6 +30,7 @@
 #include "player.h"
 #include "weapons.h"
 #include "gamerules.h"
+#include "byteswap.h"
 
 float UTIL_WeaponTimeBase( void )
 {
@@ -1781,22 +1782,52 @@ void CSave::WriteData( const char *pname, int size, const char *pdata )
 	BufferField( pname, size, pdata );
 }
 
-void CSave::WriteShort( const char *pname, const short *data, int count )
+void CSave::WriteShort( const char *pname, short *data, int count )
 {
+	for (int i = 0; i < count; i++)
+	{
+		LittleShortSW(data[i]);
+	}
+
 	BufferField( pname, sizeof(short) * count, (const char *)data );
+
+	for (int i = 0; i < count; i++)
+	{
+		LittleShortSW(data[i]);
+	}
 }
 
-void CSave::WriteInt( const char *pname, const int *data, int count )
+void CSave::WriteInt( const char *pname, int *data, int count )
 {
+	for (int i = 0; i < count; i++)
+	{
+		LittleLongSW(data[i]);
+	}
+
 	BufferField( pname, sizeof(int) * count, (const char *)data );
+
+	for (int i = 0; i < count; i++)
+	{
+		LittleLongSW(data[i]);
+	}
 }
 
-void CSave::WriteFloat( const char *pname, const float *data, int count )
+void CSave::WriteFloat( const char *pname, float *data, int count )
 {
+	for (int i = 0; i < count; i++)
+	{
+		data[i] = LittleFloat(data[i]);
+	}
+
 	BufferField( pname, sizeof(float) * count, (const char *)data );
+
+	for (int i = 0; i < count; i++)
+	{
+		data[i] = LittleFloat(data[i]);
+	}
 }
 
-void CSave::WriteTime( const char *pname, const float *data, int count )
+void CSave::WriteTime( const char *pname, float *data, int count )
 {
 	int i;
 	//Vector tmp, input;
@@ -1811,6 +1842,7 @@ void CSave::WriteTime( const char *pname, const float *data, int count )
 		if( m_pdata )
 			tmp -= m_pdata->time;
 
+		tmp = LittleFloat(tmp);
 		BufferData( (const char *)&tmp, sizeof(float) );
 		data ++;
 	}
@@ -1851,18 +1883,35 @@ void CSave::WriteString( const char *pname, const int *stringId, int count )
 #endif
 }
 
-void CSave::WriteVector( const char *pname, const Vector &value )
+void CSave::WriteVector( const char *pname, Vector &value )
 {
 	WriteVector( pname, &value.x, 1 );
 }
 
-void CSave::WriteVector( const char *pname, const float *value, int count )
+void CSave::WriteVector( const char *pname, float *value, int count )
 {
 	BufferHeader( pname, sizeof(float) * 3 * count );
+	
+	for (int i = 0; i < count; i++)
+	{
+		for (int j = 0; j < 3; j++)
+		{
+			value[j + (i * 3)] = LittleFloat(value[j + (i * 3)]);
+		}
+	}
+
 	BufferData( (const char *)value, sizeof(float) * 3 * count );
+
+	for (int i = 0; i < count; i++)
+	{
+		for (int j = 0; j < 3; j++)
+		{
+			value[j + (i * 3)] = LittleFloat(value[j + (i * 3)]);
+		}
+	}
 }
 
-void CSave::WritePositionVector( const char *pname, const Vector &value )
+void CSave::WritePositionVector( const char *pname, Vector &value )
 {
 	if( m_pdata && m_pdata->fUseLandmark )
 	{
@@ -1886,6 +1935,10 @@ void CSave::WritePositionVector( const char *pname, const float *value, int coun
 		if( m_pdata && m_pdata->fUseLandmark )
 			tmp = tmp - m_pdata->vecLandmarkOffset;
 
+		tmp.x = LittleFloat(tmp.x);
+		tmp.y = LittleFloat(tmp.y);
+		tmp.z = LittleFloat(tmp.z);
+
 		BufferData( (const char *)&tmp.x, sizeof(float) * 3 );
 		value += 3;
 	}
@@ -2037,7 +2090,9 @@ int CSave::WriteFields( const char *pname, void *pBaseData, TYPEDESCRIPTION *pFi
 			WriteInt( pTest->fieldName, (int *)pOutputData, pTest->fieldSize );
 			break;
 		case FIELD_SHORT:
+			LittleShortSW(*(short *)pOutputData);
 			WriteData( pTest->fieldName, 2 * pTest->fieldSize, ( (char *)pOutputData ) );
+			LittleShortSW(*(short *)pOutputData);
 			break;
 		case FIELD_CHARACTER:
 			WriteData( pTest->fieldName, pTest->fieldSize, ( (char *)pOutputData ) );
@@ -2086,6 +2141,9 @@ void CSave::BufferHeader( const char *pname, int size )
 	short hashvalue = TokenHash( pname );
 	if( size > 1 << ( sizeof(short) * 8 ) )
 		ALERT( at_error, "CSave :: BufferHeader() size parameter exceeds 'short'!\n" );
+
+	LittleLongSW(size);
+	LittleShortSW(hashvalue);
 	BufferData( (const char *)&size, sizeof(short) );
 	BufferData( (const char *)&hashvalue, sizeof(short) );
 }
@@ -2149,11 +2207,13 @@ int CRestore::ReadField( void *pBaseData, TYPEDESCRIPTION *pFields, int fieldCou
 					case FIELD_TIME:
 					#if __VFP_FP__
 						memcpy( &timeData, pInputData, 4 );
+						timeData = LittleFloat(timeData);
 						// Re-base time variables
 						timeData += time;
 						memcpy( pOutputData, &timeData, 4 );
 					#else
 						timeData = *(float *)pInputData;
+						timeData = LittleFloat(timeData);
 						// Re-base time variables
 						timeData += time;
 						*( (float *)pOutputData ) = timeData;
@@ -2161,6 +2221,7 @@ int CRestore::ReadField( void *pBaseData, TYPEDESCRIPTION *pFields, int fieldCou
 						break;
 					case FIELD_FLOAT:
 						memcpy( pOutputData, pInputData, 4 );
+						*( (float *)pOutputData ) = LittleFloat(*( (float *)pOutputData ));
 						break;
 					case FIELD_MODELNAME:
 					case FIELD_SOUNDNAME:
@@ -2195,6 +2256,7 @@ int CRestore::ReadField( void *pBaseData, TYPEDESCRIPTION *pFields, int fieldCou
 						break;
 					case FIELD_EVARS:
 						entityIndex = *( int *)pInputData;
+						LittleLongSW(entityIndex);
 						pent = EntityFromIndex( entityIndex );
 						if( pent )
 							*( (entvars_t **)pOutputData ) = VARS( pent );
@@ -2203,6 +2265,7 @@ int CRestore::ReadField( void *pBaseData, TYPEDESCRIPTION *pFields, int fieldCou
 						break;
 					case FIELD_CLASSPTR:
 						entityIndex = *( int *)pInputData;
+						LittleLongSW(entityIndex);
 						pent = EntityFromIndex( entityIndex );
 						if( pent )
 							*( (CBaseEntity **)pOutputData ) = CBaseEntity::Instance( pent );
@@ -2211,6 +2274,7 @@ int CRestore::ReadField( void *pBaseData, TYPEDESCRIPTION *pFields, int fieldCou
 						break;
 					case FIELD_EDICT:
 						entityIndex = *(int *)pInputData;
+						LittleLongSW(entityIndex);
 						pent = EntityFromIndex( entityIndex );
 						*( (edict_t **)pOutputData ) = pent;
 						break;
@@ -2218,6 +2282,7 @@ int CRestore::ReadField( void *pBaseData, TYPEDESCRIPTION *pFields, int fieldCou
 						// Input and Output sizes are different!
 						pInputData = (char*)pData + j * gInputSizes[pTest->fieldType];
 						entityIndex = *(int *)pInputData;
+						LittleLongSW(entityIndex);
 						pent = EntityFromIndex( entityIndex );
 						if( pent )
 							*( (EHANDLE *)pOutputData ) = CBaseEntity::Instance( pent );
@@ -2226,6 +2291,7 @@ int CRestore::ReadField( void *pBaseData, TYPEDESCRIPTION *pFields, int fieldCou
 						break;
 					case FIELD_ENTITY:
 						entityIndex = *(int *)pInputData;
+						LittleLongSW(entityIndex);
 						pent = EntityFromIndex( entityIndex );
 						if( pent )
 							*( (EOFFSET *)pOutputData ) = OFFSET( pent );
@@ -2235,10 +2301,13 @@ int CRestore::ReadField( void *pBaseData, TYPEDESCRIPTION *pFields, int fieldCou
 					case FIELD_VECTOR:
 						#if __VFP_FP__
 						memcpy( pOutputData, pInputData, sizeof( Vector ) );
+						((float*)pOutputData)[0] = LittleFloat(((float*)pOutputData)[0]);
+						((float*)pOutputData)[1] = LittleFloat(((float*)pOutputData)[1]);
+						((float*)pOutputData)[2] = LittleFloat(((float*)pOutputData)[2]);
 						#else
-						( (float *)pOutputData )[0] = ( (float *)pInputData )[0];
-						( (float *)pOutputData )[1] = ( (float *)pInputData )[1];
-						( (float *)pOutputData )[2] = ( (float *)pInputData )[2];
+						( (float *)pOutputData )[0] = LittleFloat(( (float *)pInputData )[0]);
+						( (float *)pOutputData )[1] = LittleFloat(( (float *)pInputData )[1]);
+						( (float *)pOutputData )[2] = LittleFloat(( (float *)pInputData )[2]);
 						#endif
 						break;
 					case FIELD_POSITION_VECTOR:
@@ -2246,27 +2315,30 @@ int CRestore::ReadField( void *pBaseData, TYPEDESCRIPTION *pFields, int fieldCou
 						{
 							Vector tmp;
 							memcpy( &tmp, pInputData, sizeof( Vector ) );
+							tmp.x = LittleFloat(tmp.x);
+							tmp.y = LittleFloat(tmp.y);
+							tmp.z = LittleFloat(tmp.z);
 							tmp = tmp + position;
 							memcpy( pOutputData, &tmp, sizeof( Vector ) );
 						}
 						#else
-						( (float *)pOutputData )[0] = ( (float *)pInputData )[0] + position.x;
-						( (float *)pOutputData )[1] = ( (float *)pInputData )[1] + position.y;
-						( (float *)pOutputData )[2] = ( (float *)pInputData )[2] + position.z;
+						( (float *)pOutputData )[0] = LittleFloat(( (float *)pInputData )[0]) + position.x;
+						( (float *)pOutputData )[1] = LittleFloat(( (float *)pInputData )[1]) + position.y;
+						( (float *)pOutputData )[2] = LittleFloat(( (float *)pInputData )[2]) + position.z;
 						#endif
 						break;
 					case FIELD_BOOLEAN:
 					case FIELD_INTEGER:
-						*( (int *)pOutputData ) = *(int *)pInputData;
+						*( (int *)pOutputData ) = LittleLong(*(int *)pInputData);
 						break;
 					case FIELD_SHORT:
-						*( (short *)pOutputData ) = *(short *)pInputData;
+						*( (short *)pOutputData ) = LittleShort(*(short *)pInputData);
 						break;
 					case FIELD_CHARACTER:
 						*( (char *)pOutputData ) = *(char *)pInputData;
 						break;
 					case FIELD_POINTER:
-						*( (void**)pOutputData ) = *(void **)pInputData;
+						*( (void**)pOutputData ) = (void*)LittleLong(*(int *)pInputData);
 						break;
 					case FIELD_FUNCTION:
 						if( ( (char *)pInputData )[0] == '\0' )
@@ -2354,6 +2426,8 @@ short CRestore::ReadShort( void )
 
 	BufferReadBytes( (char *)&tmp, sizeof(short) );
 
+	LittleShortSW(tmp);
+
 	return tmp;
 }
 
@@ -2363,6 +2437,8 @@ int CRestore::ReadInt( void )
 
 	BufferReadBytes( (char *)&tmp, sizeof(int) );
 
+	LittleLongSW(tmp);
+
 	return tmp;
 }
 
@@ -2371,7 +2447,7 @@ int CRestore::ReadNamedInt( const char *pName )
 	HEADER header;
 
 	BufferReadHeader( &header );
-	return ( (int *)header.pData )[0];
+	return LittleLong(( (int *)header.pData )[0]);
 }
 
 char *CRestore::ReadNamedString( const char *pName )
